backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/LibvirtVmXmlBuilder.java

```diff
@@ -1330,30 +1328,28 @@ private final void updateBootOrder(List<VmDevice> ... bootableDevices) {
     }
 
     void writeVGpu() {
-        for (String mdevType : MDevTypesUtils.getMDevTypes(vm)) {
+        for (Map<String, Object> mdevSpecParams : MDevTypesUtils.getMdevs(vm, vmInfoBuildUtils.getVmDevices(vm.getId()))) {
             writer.writeStartElement("hostdev");
             writer.writeAttributeString("mode", "subsystem");
             writer.writeAttributeString("type", "mdev");
             writer.writeAttributeString("model", "vfio-pci");
-            if (mdevDisplayOn) {
-                // Nvidia vGPU VNC console is only supported on RHEL >= 7.6
-                // See https://bugzilla.redhat.com/show_bug.cgi?id=1633623 for details and discussion
+            if (!(Boolean)mdevSpecParams.get(MDevTypesUtils.NODISPLAY)) {
```

Comment:
better to use !Boolean.TRUE.equals(...) to also cover the case NODISPLAY doesn't exist